sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosClientBuilderTest.java

```diff
@@ -207,7 +219,14 @@ public void validateSessionTokenCapturingForAccountDefaultConsistency() {
             throw new SkipException("This test is only applicable when default account-level consistency is Session.");
         }
 
-        SessionContainer sessionContainer = (SessionContainer)documentClient.getSession();
+        ISessionContainer sessionContainer = documentClient.getSession();
+
+        if (System.getProperty("COSMOS.SESSION_CAPTURING_TYPE").equals("REGION_SCOPED")) {
+            assertThat(sessionContainer instanceof RegionScopedSessionContainer).isTrue();
+        }
+
         assertThat(sessionContainer.getDisableSessionCapturing()).isEqualTo(false);
+
+        System.clearProperty("COSMOS.SESSION_CAPTURING_TYPE");
```

Comment:
Property should be cleared in finally - otherwise assert failure above would leave the process in bad state?

Reply:
Makes sense - fixed in next iteration.